### PR TITLE
Build shared libraries, add run-export

### PR DESCRIPTION
**Pull request by h-vetinari**
Towards #38

Closes #41

### CODE REVIEW

**Comment by h-vetinari** [comment]
@lidavidm @wjones127
This should be ready to go; it's now according to conda-forge default standards (only shared libs), with a run-export, and with a bunch of clean-ups. PTAL :)

**Review by wjones127 — APPROVED** [review]
Well done! The tests look quite thorough. Hopefully I can get the share libraries fixed upstream so we don't need that patch.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1 @@
-* @lidavidm
\ No newline at end of file
+* @h-vetinari @lidavidm
```
[the remark this answers is not in the document]
Comment (h-vetinari) [reply .github/CODEOWNERS:1]: That warning is just GitHub being silly. Merging the PR is all that's necessary to join as a maintainer.

File: recipe/bld.bat [hunk 2]
```diff
@@ -15,6 +15,8 @@ cmake .. ^
       -DCMAKE_CXX_STANDARD=17 ^
       -DCMAKE_PREFIX_PATH=%CONDA_PREFIX% ^
       -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
+      -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON ^
```
Comment (lidavidm) [inline recipe/bld.bat:18]: Seems like this may need to be fixed upstream (presumably dllimport/dllexport annotations are missing)?

**Review by lidavidm — APPROVED** [review]
Thank you!

File: recipe/bld.bat (hunk 2 above)
Comment (h-vetinari) [reply recipe/bld.bat:18]: Yes, but those are really invasive and that's why many projects don't have them 🤷

**Comment by h-vetinari** [comment]
Thanks for merging! :)

File: recipe/bld.bat (hunk 2 above)
Comment (lidavidm) [reply recipe/bld.bat:18]: more practically, google never fixed the protobuf compiler to be able to insert them properly so I guess it's not possible to fix anyways. (at least the last time I took a swing at that.)

**Comment by h-vetinari** [comment]
> Thanks for merging! :)

... though I wish you hadn't squashed everything. 🤷 Makes it harder to do archeology etc. on the feedstock.

**Comment by lidavidm** [comment]
Ah sorry, it appears to default to squash for me and usually I just go with that. I'll try to keep that in mind.

**Pull request merged**